docs/proposals/proposal-trigger-and-deploy.md

```diff
@@ -0,0 +1,286 @@
+
+Multiple deployments will produce inaccurate results as we can only measure
+a single project per node. We can set concurrency in the workflow to ensure
+only a single instance runs at a time.
+
+Deployment may fail. What alerting do we need? Do we also need to notify the
+project?
+
+Uninstall at end of pipeline fails. We can wait till all flux finalizers are
+removed. In future we could create nodes on demand and delete on completion.
+
+## Design Details
+
+### Subscribing to Releases
+
+A YAML file of CNCF projects and their latest release will be stored in the
+tooling repo e.g.
+
+```yaml
+projects:
+  - name: falco
+    feed_url: "https://github.com/falcosecurity/falco/releases.atom"
+    latest_release: "0.37.1"
+    sub_components:
+      - ebpf
+      - modern-ebpf
+      - kmod
+```
+
+A scheduled GitHub Action will run every 30 minutes and check the Atom feed of
+each project for new releases.
+
+If a new release is detected the action will trigger the pipeline for the new
+release and update the YAML file with the new version. This is to ensure each
+release is only triggered once.
+
+If sub components are specified then the pipeline will be triggered once per
+sub component.
+
+### Trigger
+
+The green reviews pipeline will be triggered by sending a [workflow_dispatch](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#workflow_dispatch)
+event via its GitHub webhook.
+
+Inputs are
+
+- `cncf_project`: **required** Project to be deployed e.g. `falco`
+- `cncf_project_sub`: **optional** Subcomponent if project has multiple variants
+they wish to test e.g. `modern-ebpf`
+- `version`: **required** Version of project to be tested e.g. `0.37.0`
+
+```sh
+curl -X POST \
+     -H "Accept: application/vnd.github.v3+json" \
+     -H "Authorization: token $GITHUB_PAT" \
+     https://api.github.com/repos/cncf-tags/green-reviews-tooling/actions/workflows/pipeline.yaml/dispatches \
+     -d '{"ref":"main", "inputs": {"cncf_project": "falco", "cncf_project_sub": "modern-ebpf","version":"0.37.0"}}'
+```
+
+The CNCF projects will be given a GitHub fine grained access token limited to
+the `green-reviews-tooling` repo. This token will have
+
+- Read access to code and metadata
+- Read write access to actions
+
+### Deploy
+
+Flux will be used to deploy the CNCF project. Projects are able to use either
+`kustomization` or `helmrelease` resources to deploy their project.
+
+Project resources that should always be deployed in the cluster are stored in
+the current location in the tooling repo, below are format where we might store
+project related configurations
+```
+# for the cncf_project
+clusters/projects/${project_name}
+
+# and for each cncf_project (different configurations)
+clusters/projects/${project_name}/${configuration_name}
+```
+
+e.g. `/clusters/projects/falco/`
+and are reconciled by source-controller.
+
+When the pipeline executes it will look for yaml files in the projects dir.
```

Comment:
What is the use of these yaml files? The project is already deployed using flux.

Reply:
For the first implementation all components including Falco and the benchmarks are deployed 24/7 by Flux.

We want to change that so the flux resources only exist for the lifetime of the workflow run. This will allow us to test more configurations of Falco and later other CNCF projects using the same cluster nodes.

Thanks that's great feedback that this isn't clear. I'll rework this section.